reviews/srv/reviews-service.js

```diff
@@ -1,25 +1,24 @@
 const cds = require ('@sap/cds')
-module.exports = cds.service.impl (function(){
+module.exports = cds.service.impl (async function(){
 
   // Get the CSN definition for Reviews from the db schema for sub-sequent queries
   // ( Note: we explicitly specify the namespace to support embedded reuse )
   const { Reviews, Likes } = this.entities ('sap.capire.reviews')
+  const messaging = await cds.connect.to('messaging')
 
   this.before (['CREATE','UPDATE'], 'Reviews', req => {
     if (!req.data.rating) req.data.rating = Math.round(Math.random()*4)+1
   })
 
   // Emit an event to inform subscribers about new avg ratings for reviewed subjects
-  // ( Note: req.on.succeeded ensures we only do that if there's no error )
   this.after (['CREATE','UPDATE','DELETE'], 'Reviews', async(_,req) => {
     const {subject} = req.data
     const {rating} = await cds.transaction(req) .run (
       SELECT.one (['round(avg(rating),2) as rating']) .from (Reviews) .where ({subject})
     )
-    req.on ('succeeded', ()=>{
-      global.it || console.log ('< emitting:', 'reviewed', { subject, rating })
-      this.emit ('reviewed', { subject, rating })
-    })
+    const msgTx = messaging.tx(req)
+    global.it || console.log ('< emitting:', 'reviewed', { subject, rating })
+    msgTx.emit ('review/reviewed', { subject, rating })
```

Comment:
```suggestion
    global.it || console.log ('< emitting:', 'reviewed', { subject, rating })
    messaging.tx(req).emit ('review/reviewed', { subject, rating })
```

Reply:
Hi Daniel,

Sure, we can do that. I chose `review/reviewed` because the message broker is shared among all involved microservices. That would at least reduce the risk of name clashes. But I agree that `reviewed` is better on the eyes here.